src/Confluent.Kafka/Serialization/StringDeserializer.cs

```diff
@@ -26,32 +30,93 @@ public class StringDeserializer : IDeserializer<string>
         Encoding encoding;
 
         /// <summary>
+        ///     Name of the encoding configuration parameter for the case of key deserialization.
+        /// </summary>
+        public const string KeyEncodingConfigParam = "dotnet.string.deserializer.encoding.key";
+
+        /// <summary>
+        ///     Name of the encoding configuration parameter for the case of value deserialization.
+        /// </summary>
+        public const string ValueEncodingConfigParam = "dotnet.string.deserializer.encoding.value";
+
+
+        /// <summary>
         ///     Initializes a new StringDeserializer class instance.
         /// </summary>
         /// <param name="encoding">
-        ///     The encoding to use when deserializing.
+        ///     The encoding to use when deserializing. For available options, refer to:
```

Comment:
well, the extra cases are just kind of for convenience because people who didn't read the documentation will probably try to use them (and I didn't see any reason for this not to work). they are all duplicates of options provided in this list. i didn't think this was worth mentioning - there is nothing people can do that isn't documented here.